### ISSUE TITLE
Request for adding generation of 'switch' statements

### ISSUE THREAD

**Issue by ZviRackover** [csmith-project/csmith#10]
This is a feature request for adding generation of switch statements.
I am adding it because:
1. I would find this feature useful
2. I would like to know if there is any work in progress towards implementing this feature, in case i decide to implement it myself

**Comment by eeide** [comment]
Hi! I don't believe that anybody is currently working on this. You could be the first!

**Comment by regehr** [comment]
I agree this would be useful, and I had totally forgotten that we never got around to it. I'm not sure why.

One thing to keep in mind is that compilers have many ways of optimizing switch statements and it would be great to figure out how to stress test all of them. So we want dense switches, sparse ones, clumpy dense ones, fallthroughs, etc.

**Comment by regehr** [comment]
I'll add that I doubt this is too difficult. Maybe Yang or Xuejun could chime in with some thoughts?

**Comment by jxyang** [comment]
I think it's conceptually a variation of if-else statement. On one hand,
the condition is simplified to a single variable, on the other hand, there
are more branches, requiring a more complex merge of after-statement facts.
It would stress the program analysis to some extend.

I would base the implementation on class StatementIf.

-Xuejun

**Comment by ZviRackover** [comment]
@jxyang Yes, I definitely agree that 'switch' being a generalization of 'if', should have some relationship. I was thinking of deriving both StatementIf and StatementSwitch from a (abstract?) base class StatementBranch.
I am new to csmith so i'm not sure if this makes sense, so please let me know if this is not a good direction.

**Comment by jxyang** [comment]
Sure. That makes sense to me.
